From 5f7825193253a3a8eb873c58717146f66a052781 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Sat, 10 Oct 2020 01:54:13 -0400
Subject: [PATCH 3/5] setup_cmake_args

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index a3f7bbe0..99f40ab2 100644
--- a/setup.py
+++ b/setup.py
@@ -141,14 +141,13 @@ class CMakeBuild(build_ext):
                       '-DPYTHON_EXECUTABLE=' + sys.executable]
 
         cmake_args += cmake_extra_options 
+        cmake_args += os.environ.get("DLIB_CMAKE_ARGS", "").strip().splitlines()
 
         cfg = 'Debug' if self.debug else 'Release'
         build_args = ['--config', cfg]
 
         if platform.system() == "Windows":
             cmake_args += ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}'.format(cfg.upper(), extdir)]
-            if sys.maxsize > 2**32:
-                cmake_args += ['-A', 'x64']
             # Do a parallel build
             build_args += ['--', '/m'] 
         else:
-- 
2.17.1

